language/move-lang/src/errors/diagnostic_codes.rs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use codespan_reporting_new::diagnostic::Severity;
+
+//**************************************************************************************************
+// Main types
+//**************************************************************************************************
+
+#[derive(PartialEq, Eq, Clone, Debug, Hash)]
+pub(crate) struct DiagnosticInfo {
+    pub(crate) severity: Severity,
+    category: Category,
+    code: u8,
+    message: &'static str,
+}
+
+pub(crate) trait DiagnosticCode: Copy {
+    fn severity() -> Severity;
+
+    fn category() -> Category;
+
+    fn code_and_message(self) -> (u8, &'static str);
```

Comment:
Not sure if you want to add this now, but eventually I think we'll want to add a `fn description() -> &'static str` that can be used by the error explanation tooling for these error codes at a later date.

Reply:
maybe... but for another date I think. Should be easy enough to add later